src/gei/Commands/MigrateRepoCommand.cs

```diff
@@ -454,6 +472,54 @@ private void LogAndValidateOptions(MigrateRepoCommandArgs args)
                 _log.LogInformation($"GIT ARCHIVE URL: {args.GitArchiveUrl}");
                 _log.LogInformation($"METADATA ARCHIVE URL: {args.MetadataArchiveUrl}");
             }
+            if (args.LfsMappingFile != "")
+            {
+                _log.LogInformation($"LFS MAPPING FILE: {args.LfsMappingFile}");
+            }
+        }
+
+        private byte[] applyLfsMappingFileToMetadata(byte[] metadataArchiveContent, string lfsMappingFile)
+        {
+            // extract the metadata archive
+            var gzipDataStream = new MemoryStream(metadataArchiveContent);
+            Stream inStream = new GZipInputStream(gzipDataStream);
+            TarArchive archive = TarArchive.CreateInputTarArchive(inStream, TarBuffer.DefaultBlockFactor);
+            archive.ExtractContents("./archiveExtracted");
+            
+            // modify the pull_requests_*.json files in the extracted archive
+            string[] fileNames = System.IO.Directory.GetFiles("./archiveExtracted");
+            foreach (string fileName in fileNames)
+            {
+                string text = File.ReadAllText(fileName);
+                var lfsMappingLines = File.ReadLines(lfsMappingFile);
+                foreach (var lfsMappingLine in lfsMappingLines)
+                {
+                    string[] lfsMapping = lfsMappingLine.Split(','); 
+                    text = text.Replace(lfsMapping[0], lfsMapping[1]);
+                }
+                File.WriteAllText(fileName, text);
+            }
```

Comment:
1. This is a bit inefficient because you traverse the entire file for each LFS mapping where as it'd be better to go through each file once and perform the mapping.
2. For each file in the extracted archive directory you load the LFS mapping file and do the split logic! It's be way more optimized to read the LFS mapping file once and create a `Dictionary` (aka hash map) and then for each file, traverse it line by line and do the mapping.